.github/workflows/build-package.yml

```diff
@@ -2,37 +2,68 @@ name: Build package
 
 on:
   workflow_call:
-    inputs:
-      container-registry:
-        required: true
-        type: string
-      container-name:
-        required: true
-        type: string
-      package-name:
-        required: true
-        type: string
-      compiler-name:
-        required: true
-        type: string
-      compiler-version:
-        required: true
-        type: string
-      spack-packages-version:
-        required: false
-        type: string
-        description: Either a git tag or branch name for the ACCESS-NRI/spack_packages repository, which defaults to the main branch
-        default: main
 
 jobs:
-  build:
+  setup-spack-packages:
+    name: Get information from spack_packages
     runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get-version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: access-nri/spack_packages
+          ref: main
+          fetch-tags: true
+      - name: Get latest spack_packages tags
+        id: get-version
+        run: echo "version=$(git describe --tags)" >> $GITHUB_OUTPUT
 
+  setup-build-ci:
+    name: Get information from build-ci
+    runs-on: ubuntu-latest
+    outputs:
+      compilers: ${{ steps.get-compilers.outputs.compilers }}
+      model: ${{ steps.get-model.outputs.model }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: access-nri/build-ci
+    - name: Get compilers to test
+      id: get-compilers
+      run: echo "compilers=$(jq -c . containers/compilers.json)" >> $GITHUB_OUTPUT
+    - name: Get model to test
+      id: get-model
+      # model-components are associated with an overarching model (for example, cice5 is associated with access-om2), this uses models.json to find the associated model
+      run: echo "model=$(jq -c 'to_entries[] | select(.value | contains([${{ github.event.repository.name }}]))' containers/models.json)" >> $GITHUB_OUTPUT
```

Comment:
As we discussed. There *might* be a better way of doing this. But I wouldn't worry. If it works, it works. Not like this will get a million hits an hour :)